### DISCUSSION TITLE
Nixie Circuit Groups Acting Peculiar

### DISCUSSION THREAD

**Discussion by rock-crusher** [tagyoureit/nodejs-poolController#419] [Nixie] [answered]
Hi guys,
Well, after my Pi4 died, I finally got a new one.
I got it set up and am once again experimenting with it. As you know I am a complete novice at this and quite frankly struggle along with trying to figure things out using Nixie. I’ve been able to get most working again, pump via RS485, valves, lighting, schedules etc. figured out, but have stumbled on what should be a very simple issue to fix, problem is, I'm just not savvy enough to figure it out on my own.

Issue 1: When creating circuit groups (lighting for example) they don’t operate as I think they should.
Example: In my experiment I created a group called "yard lighting"; it consists of pool light, spa light, tree lights, and post lights.
 
![light group](https://user-images.githubusercontent.com/94593954/163696482-08f13a45-97db-4cb0-8ef7-934868054225.JPG)

This shows up as a feature button on the dashboard. 
When pressed, everything turns on as expected via REM/Sequent relay board and each light turns off automatically according to its individual egg timer setting (all is good). 

![yard lights](https://user-images.githubusercontent.com/94593954/163696541-ee7da056-6ef3-47d5-9fb2-e238aa315d4d.JPG)

![Relay Board](https://user-images.githubusercontent.com/94593954/163696546-c9673d19-87c8-4855-8ef3-a9c7fb2b2b63.JPG)

Now, if I press the "yard lighting" button again before all lights have timed out and shut off (I'm expecting all lights to turn off) only one light turns off and it is always the last one listed in the circuit group. I have shifted the lights around in the group just to see what happens and sure enough the only one that will turn off is the last light listed in the circuit group. The others stay on indefinitely even though the dashboard indicates all lights have turned off. The reality is only one turned off and I must then go to REM/Sequent relay board and click each of them off manually.

![yard lights off](https://user-images.githubusercontent.com/94593954/163696606-5ea43683-e481-49a8-b377-967cb0264500.JPG)
![relay board after yard lights turned off](https://user-images.githubusercontent.com/94593954/163696612-34203c41-0d21-43c6-a71f-b62881f3e092.JPG)

Issue 2: When making a schedule for the same light group ("yard lighting") the group of lights turns on according to the "on" schedule as it should, but when the "off" time arrives just two of the lights will turn off. The two lights that turn off are random each time I try it. I then must go back into REM/Sequent relay board and click each of them off manually.

So, what you all think, what the heck is going on here?  

Thanks in advance.

**Comment by rstrouse** [comment, marked as answer]
Pull njsPC.  There was some weirdness with the process where it was still waiting on the latch to expire.

**Reply by rock-crusher** [reply]
Thank you rstrouse for correcting the issue, it now seems to be working as it should. 😎
